src/analysis/normed/group/basic.lean

```diff
@@ -1136,6 +1136,27 @@ def normed_group.induced {E} [add_comm_group E]
 { .. semi_normed_group.induced f,
   .. metric_space.induced f h normed_group.to_metric_space, }
 
+/-- The constant norm that is `c` everywhere except at `dist x x = 0`.
+
+See note [reducible non-instances]-/
+@[reducible]
+def normed_group.const {E} [decidable_eq E] [add_comm_group E] (c : ℝ) (hc : 0 < c) :
+  normed_group E :=
+{ norm := λ x, if x = 0 then 0 else c,
+  dist_eq := λ x y, if_congr sub_eq_zero.symm rfl rfl,
+  to_metric_space := metric_space.const ⟨c, hc.le⟩ $ (nonneg.mk_eq_zero _).not.mpr hc.ne' }
+
+lemma const_norm_eq {E} [decidable_eq E] [add_comm_group E] (c : ℝ) (hc : 0 < c)
```

Comment:
What about a lemma that assumes `x ≠ 0` and states `norm x = c`?

Reply:
Good idea